tools/webpack/modules.js

```diff
@@ -14,6 +14,10 @@ const {
 	WORDPRESS_NAMESPACE,
 } = require( './shared' );
 
+const SUBMODULES = [
+	{ name: 'interactivity-debug', path: 'interactivity/build-module/debug' },
+];
```

Comment:
I've created the `SUBMODULES` list because I thought that adding the `interactivity-debug` name to `MODULES` doesn't seem right:

- `MODULES` is a list of package names ( e.g., `@wordpress/interactivity`, see [this](https://github.com/WordPress/wordpress-develop/blob/add/interactivity-api-debug-mode/tools/webpack/shared.js))
- `MODULES` list is also used [here](https://github.com/WordPress/wordpress-develop/blob/dc15d66ee6203bf7b480de3c120d0d64f14009e4/tools/webpack/packages.js#L75) to filter out some packages that should be bundled as modules.

In this case, `interactivity-debug` is not a package but a _submodule_ (sort of) inside the `@wordpress/interactivity` package. Also, we need both a name and a path to define the entry point.

I guess this is better than defining the entry point directly inside `config`, but happy to change it if others think there's a better way. 🙂

Reply:
The result is its own module, we compile a separate file that's a distinct module. My recommendation would be to move this with the rest of the modules where it's visible in the same place. That also reduces the changes needed in this file.

Reply:
`MODULES` is a flat list, I missed that detail 🤔 

https://github.com/WordPress/wordpress-develop/blob/dc15d66ee6203bf7b480de3c120d0d64f14009e4/tools/webpack/shared.js#L99

Reply:
This seems fine for now as implemented.

Reply:
Can we bundle instead different files depending on the build mode: `production` vs `development`? That would simplify everything as we could keep the distinction between `.js` vs `.min.js` using a single module identifier.

Reply:
Good thought, I bet we could figure something like that out dynamically.

Reply:
Hey @gziolo, I'm not sure I follow. Do you mean changing the entry point of `@wordpress/interactivity` dynamically depending on the build mode? 🤔

Reply:
Thanks for the suggestion, @gziolo. It works as expected. 🙌 

The PR still needs the interactivity API dependencies to be updated, though. I guess they are updated following a specific process instead of manually, right?

Reply:
I think that's happening soon: https://github.com/WordPress/wordpress-develop/pull/6612

Reply:
You can rebase with the latest `trunk` to have the latest version of Interactivity API.